File: 2022/Python/04.py
from parse import * 

def generator(input) :
    return [[int(n) for n in parse("{}-{},{}-{}", line)[:4]] for line in input.splitlines()]

def part_1(input):
    return solver(input)
    
def part_2(input) :
    return solver([[a ,b, d, c] for [a, b, c, d] in input])

def solver(input) :
    return sum([(a - c) * (b - d) <= 0 for [a,b,c,d] in input])

File: 2022/04/04.rs
aoc2022::main!();

fn generator(input : &str) -> Vec<(i16, i16, i16, i16)> {
    input
        .lines()
        .map(|line| line.split(['-', ','])
            .map(|n| n.parse::<i16>().unwrap())
            .collect_tuple()
            .unwrap())
        .collect_vec()
}

fn part_1(assignements : Vec<(i16, i16, i16, i16)>) -> usize {
    solver(assignements)
}

fn part_2(assignements : Vec<(i16, i16, i16, i16)>) -> usize {
    solver(assignements.iter().map(|&(a,b,c,d)| (a,b,d,c)).collect_vec())
}

fn solver(input :  Vec<(i16, i16, i16, i16)>) -> usize{
    input
        .iter()
        .map(|(a, b, c, d)| ((a - c) * (b - d) <= 0) as usize)
        .sum()
}